### DISCUSSION TITLE
Arbitrary values, screen import and important prefix

### DISCUSSION THREAD

**Discussion by ben-rogerson** [ben-rogerson/twin.macro#451] [Releases]
## New: Arbitrary values (#447)

Twin now supports the same arbitrary values syntax popularized by [Tailwind’s jit ("Just-in-Time") mode](https://tailwindcss.com/docs/just-in-time-mode#enabling-jit-mode).

```js
tw`top-[calc(100vh - 2rem)]`

// ↓ ↓ ↓ ↓ ↓ ↓

;({ top: 'calc(100vh - 2rem)' })
```

This is a great way to add once-off values that don’t need to be in your tailwind config.

[Read more about arbitrary values →](https://github.com/ben-rogerson/twin.macro/blob/master/docs/arbitrary-values.md)

## New: Important prefix (#449)

Also from Tailwind JIT, is the bang `!` prefix to add important styles: 

```js
tw`!block`;

// ↓ ↓ ↓ ↓ ↓ ↓

;({ "display": "block !important" })
```

Twin will continue to support a bang at the end of the class too - so you've got a couple of options.

[See the pr for more info →](https://github.com/ben-rogerson/twin.macro/pull/449)

## New: Screen import (#408)

The screen import creates media queries for custom css that sync with your tailwind config screen values (sm, md, lg, etc).

**Usage with the css prop**

```js
import tw, { screen, css } from 'twin.macro'

const styles = [
  screen`sm`({ display: 'block', ...tw`inline` }),
]

<div css={styles} />
```

**Usage with styled components**

```js
import tw, { styled, screen, css } from 'twin.macro'

const Component = styled.div(() => [
  screen`sm`({ display: 'block', ...tw`inline` }),
])

<Component />
```

[Read more about the screen import →](https://github.com/ben-rogerson/twin.macro/blob/master/docs/screen-import.md)

## Bugfixes

- Fix values issues on `[divide/space]-[x-/y-]reverse` (#359)

<hr /><em>This discussion was created from the release <a href='https://github.com/ben-rogerson/twin.macro/releases/tag/2.5.0'>Arbitrary values, screen import and important prefix</a>.</em>

**Comment by luknl** [comment]
Thank you so much for the great work! Twin really rocks 👏
